clang/test/SemaSYCL/loop_fusion.cpp

```diff
@@ -0,0 +1,102 @@
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -internal-isystem %S/Inputs -Wno-sycl-2017-compat -ast-dump -verify %s | FileCheck %s
+
+#include "sycl.hpp"
+
+using namespace cl::sycl;
+queue q;
+
+// CHECK: FunctionDecl {{.*}} func1 'void ()'
+// CHECK-NEXT: CompoundStmt
+// CHECK-NEXT: SYCLIntelLoopFuseAttr
+// CHECK-NEXT: NULL
+[[intel::loop_fuse]] void func1() {}
+
+// CHECK: FunctionDecl {{.*}} func2 'void ()'
+// CHECK-NEXT: CompoundStmt
+// CHECK-NEXT: SYCLIntelLoopFuseAttr
+// CHECK-NEXT: ConstantExpr
+// CHECK-NEXT: value: Int 0
+[[intel::loop_fuse(0)]] void func2() {}
+
+// CHECK: FunctionDecl {{.*}} func3 'void ()'
+// CHECK-NEXT: CompoundStmt
+// CHECK-NEXT: SYCLIntelLoopFuseIndependentAttr
+// CHECK-NEXT: NULL
+[[intel::loop_fuse_independent]] void func3() {}
+
+// CHECK: FunctionDecl {{.*}} func4 'void ()'
+// CHECK-NEXT: CompoundStmt
+// CHECK-NEXT: SYCLIntelLoopFuseIndependentAttr
+// CHECK-NEXT: ConstantExpr
+// CHECK-NEXT: value: Int 3
+[[intel::loop_fuse_independent(3)]] void func4() {}
+
+class KernelFunctor {
+public:
+  void operator()() const {
+    func1();
+    func3();
+  }
+};
+
+template <int N>
+class KernelFunctor2 {
+public:
+  [[intel::loop_fuse(N)]] void operator()() const {
+  }
+};
+
+void foo() {
+  q.submit([&](handler &h) {
+    // CHECK: FunctionDecl {{.*}}kernel_name_1 'void ()'
+    // CHECK-NOT: SYCLIntelLoopFuseAttr
+    // CHECK-NOT: SYCLIntelLoopFuseIndependentAttr
+    KernelFunctor f1;
+    h.single_task<class kernel_name_1>(f1);
+
+    // CHECK: FunctionDecl {{.*}}kernel_name_2 'void ()'
+    // CHECK: SYCLIntelLoopFuseAttr
+    // CHECK-NEXT: SubstNonTypeTemplateParmExpr
+    // CHECK-NEXT: NonTypeTemplateParmDecl
+    // CHECK-NEXT: IntegerLiteral {{.*}} 'int' 3
+    KernelFunctor2<3> f2;
+    h.single_task<class kernel_name_2>(f2);
+
+    // CHECK: FunctionDecl {{.*}}kernel_name_3 'void ()'
+    // CHECK: SYCLIntelLoopFuseIndependentAttr
+    h.single_task<class kernel_name_3>(
+        []() [[intel::loop_fuse_independent]]{});
+  });
+
+  [[intel::loop_fuse]] int testVar = 0; // expected-error{{'loop_fuse' attribute only applies to functions}}
+}
+
+[[intel::loop_fuse(1048577)]] void func5() {}        // expected-error{{'loop_fuse' attribute requires integer constant between 0 and 1048576 inclusive}}
+[[intel::loop_fuse_independent(-1)]] void func6() {} // expected-error{{'loop_fuse_independent' attribute requires integer constant between 0 and 1048576 inclusive}}
```

Comment:
Can you also add a test for `[[intel::loop_fuse(0, 1)]]` to validate that it only accepts zero or one arguments?